packages/discovery-provider/ddl/functions/handle_playlist.sql

```diff
@@ -86,8 +87,29 @@ begin
 
   begin
     if new.is_delete IS FALSE and new.is_private IS FALSE then
-      for track_item IN select jsonb_array_elements from jsonb_array_elements(new.playlist_contents -> 'track_ids')
+      for track_item IN select jsonb_array_elements from jsonb_array_elements(new.playlist_contents->'track_ids')
```

Comment:
is there a way to use set-based operations to avoid nesting loops? Maybe something as follows:

1. Select all the buyer_user_ids:
```
        WITH album_purchasers AS (
            SELECT DISTINCT buyer_user_id
            FROM usdc_purchases
            WHERE content_id = NEW.playlist_id
            AND content_type = 'album'
        )
```
2. join this with the notification section, kinda like this
```
insert into notification ....
select new.blocknumber,
       ARRAY [album_purchaser.buyer.user_id]
       ....
join album_purchasers as album_purchaser
```

Reply:
Great call! We've updated the query and tested. It correctly writes a row per user who purchased all in a single insert op.